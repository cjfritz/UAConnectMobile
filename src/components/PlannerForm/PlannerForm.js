
import React, { Component } from 'react';
import { View } from 'react-native';
import { connect } from 'react-redux';
import {
  Item, H3, Input, Icon,
} from 'native-base';
import { plannerUpdate, plannerValidUpdate } from '../../actions/PlannerActions';
import FormPicker from '../common/FormPicker/FormPicker';
import styles from './PlannerForm.style';

export class PlannerForm extends Component {
  getCourseAndDescription = () => {
    const { course, description } = this.props;
    return `${course} - ${description}`;
  };

  getTermType = () => {
    const { term } = this.props;
    const termSplit = term.split(' - ');
    return termSplit[0];
  };

  getTermYear = () => {
    const { term } = this.props;
    const termSplit = term.split(' - ');
    return termSplit[1];
  };

  setCourseAndDescription = value => {
    const { updatePlannerForm, updateValidFields } = this.props;
    const split = value.split(' - ');
    updatePlannerForm({ prop: 'course', value: split[0] });
    updateValidFields({ prop: 'course', value: split[0] });
    updatePlannerForm({ prop: 'description', value: split[1] });
    updateValidFields({ prop: 'description', value: split[1] });
  };

  setTermType = value => {
    const { updatePlannerForm, updateValidFields, term } = this.props;
    const termSplit = term.split(' - ');
    updatePlannerForm({ prop: 'term', value: `${value} - ${termSplit[1]}` });
    updateValidFields({ prop: 'term', value: `${value} - ${termSplit[1]}` });
  };

  setTermYear = value => {
    const { updatePlannerForm, updateValidFields, term } = this.props;
    const termSplit = term.split(' - ');
    updatePlannerForm({ prop: 'term', value: `${termSplit[0]} - ${value}` });
    updateValidFields({ prop: 'term', value: `${termSplit[0]} - ${value}` });
  };

  setGrade = text => {
    const { updatePlannerForm, updateValidFields } = this.props;
    updatePlannerForm({ prop: 'grade', value: text });
    updateValidFields({ prop: 'grade', value: text });
  };

  setUnits = text => {
    const { updatePlannerForm, updateValidFields } = this.props;
    updateValidFields({ prop: 'units', value: text });
    updatePlannerForm({ prop: 'units', value: text });
  };

  render() {
    const {
      grade,
      units,
      validFields,
      mockCourses,
      terms,
      termYears,
    } = this.props;
    return (
      <View style={ styles.container }>
        <H3 style={ styles.headerText }>
          Choose Course:
        </H3>
        <FormPicker
          placeholder='Select Course'
          selectedValue={ this.getCourseAndDescription }
          onValueChange={ this.setCourseAndDescription }
          items={ mockCourses }
        />
        <H3 style={ styles.headerText }>
          Choose Term Type:
        </H3>
        <FormPicker
          placeholder='Select Term Type'
          selectedValue={ this.getTermType }
          onValueChange={ this.setTermType }
          items={ terms }
        />
        <H3 style={ styles.headerText }>
          Choose Term Year:
        </H3>
        <FormPicker
          placeholder='Select Term Year'
          selectedValue={ this.getTermYear }
          onValueChange={ this.setTermYear }
          items={ termYears }
        />
        <H3 style={ styles.headerText }>
          GPA Earned:
        </H3>
        <Item error={ !validFields.validGrade }>
          <Input
            style={ styles.input }
            placeholder='Enter Grade'
            keyboardType='decimal-pad'
            onChangeText={ text => this.setGrade(text) }
            value={ grade }
          />
          { validFields.validGrade ? <Icon name='checkmark-circle' />
            : <Icon name='close-circle' />
          }
        </Item>
        <H3 style={ styles.headerText }>
          Units Earned:
        </H3>
        <Item error={ !validFields.validUnits }>
          <Input
            style={ styles.input }
            placeholder='Enter Units'
            keyboardType='number-pad'
            onChangeText={ text => this.setUnits(text) }
            value={ units }
          />
          { validFields.validUnits ? <Icon name='checkmark-circle' />
            : <Icon name='close-circle' />
          }
        </Item>
      </View>
    );
  }
}

const mapDispatchToProps = dispatch => (
  ({
    updatePlannerForm: ({ prop, value }) => dispatch(plannerUpdate({ prop, value })),
    updateValidFields: ({ prop, value }) => dispatch(plannerValidUpdate({ prop, value })),
  })
);

const mapStateToProps = state => {
  const {
    course,
    description,
    term,
    grade,
    units,
    validFields,
    mockCourses,
    terms,
    termYears,
  } = state.plannerForm;
  return {
    course,
    description,
    term,
    grade,
    units,
    validFields,
    mockCourses,
    terms,
    termYears,
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(PlannerForm);
