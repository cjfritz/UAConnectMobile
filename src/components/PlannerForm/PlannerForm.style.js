import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  container: {
    alignContent: 'center',
    backgroundColor: 'white',
    flex: 1,
  },
  headerText: {
    color: 'darkred',
    marginLeft: '4%',
    marginTop: '5%',
  },
  input: {
    fontSize: 15,
    marginLeft: '4%',
  },
  text: {
    color: 'white',
    fontSize: 20,
  },
  titleText: {
    color: 'white',
    fontSize: 20,
  },
});
