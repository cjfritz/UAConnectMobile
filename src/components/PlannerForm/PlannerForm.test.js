import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { PlannerForm } from './PlannerForm';
import { mockCourses, terms } from '../../assets/fixtures/FormFixtures';

const termYears = [2000, 2001, 2002];
const props = {
  coures: '',
  description: '',
  term: '',
  grade: '',
  units: '',
  validFields: {
    validCourse: false,
    validDescription: false,
    validTerm: false,
    validGrade: false,
    validUnits: false,
  },
  mockCourses,
  terms,
  termYears,
  updatePlannerForm: jest.fn(),
};

describe('<PlannerForm />', () => {
  it('renders with empty fields', () => {
    const shallowRenderer = new ShallowRenderer();
    shallowRenderer.render(<PlannerForm { ...props } />);
    expect(shallowRenderer.getRenderOutput()).toMatchSnapshot();
  });

  it('renders with completed fields', () => {
    const newProps = {
      coures: 'testCourse',
      description: 'testDescription',
      term: 'testTerm',
      grade: 4.0,
      units: 4,
      validFields: {
        validCourse: true,
        validDescription: true,
        validTerm: true,
        validGrade: true,
        validUnits: true,
      },
    };

    const shallowRenderer = new ShallowRenderer();
    shallowRenderer.render(<PlannerForm { ...props } { ...newProps } />);
    expect(shallowRenderer.getRenderOutput()).toMatchSnapshot();
  });
});
