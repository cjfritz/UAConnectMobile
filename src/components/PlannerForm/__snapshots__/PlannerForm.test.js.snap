// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlannerForm /> renders with completed fields 1`] = `
<Component
  style={
    Object {
      "alignContent": "center",
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <Styled(H3)
    style={
      Object {
        "color": "darkred",
        "marginLeft": "4%",
        "marginTop": "5%",
      }
    }
  >
    Choose Course:
  </Styled(H3)>
  <FormPicker
    items={
      Array [
        "CSCE 2004 - Programming Foundations I",
        "CSCE 2014 - Programming Foundations II",
        "CSCE 2114 - Digital Design",
        "CSCE 2214 - Computer Organization",
        "CSCE 3193 - Programming Paradigms",
        "CSCE 3193H - Honors Programming Paradigms",
      ]
    }
    onValueChange={[Function]}
    placeholder="Select Course"
    selectedValue={[Function]}
  />
  <Styled(H3)
    style={
      Object {
        "color": "darkred",
        "marginLeft": "4%",
        "marginTop": "5%",
      }
    }
  >
    Choose Term Type:
  </Styled(H3)>
  <FormPicker
    items={
      Array [
        "Spring Intersession",
        "Spring Semester",
        "Summer Intersession",
        "Summer 10 Week Session",
        "Summer 5 Week Session 1",
        "Summer 5 Week Session 2",
        "Summer 8 Week Session",
        "Fall Intersession",
        "Fall Semester",
      ]
    }
    onValueChange={[Function]}
    placeholder="Select Term Type"
    selectedValue={[Function]}
  />
  <Styled(H3)
    style={
      Object {
        "color": "darkred",
        "marginLeft": "4%",
        "marginTop": "5%",
      }
    }
  >
    Choose Term Year:
  </Styled(H3)>
  <FormPicker
    items={
      Array [
        2000,
        2001,
        2002,
      ]
    }
    onValueChange={[Function]}
    placeholder="Select Term Year"
    selectedValue={[Function]}
  />
  <Styled(H3)
    style={
      Object {
        "color": "darkred",
        "marginLeft": "4%",
        "marginTop": "5%",
      }
    }
  >
    GPA Earned:
  </Styled(H3)>
  <Styled(Item)
    error={false}
  >
    <Styled(Input)
      keyboardType="decimal-pad"
      onChangeText={[Function]}
      placeholder="Enter Grade"
      style={
        Object {
          "fontSize": 15,
          "marginLeft": "4%",
        }
      }
      value={4}
    />
    <Styled(Icon)
      name="checkmark-circle"
    />
  </Styled(Item)>
  <Styled(H3)
    style={
      Object {
        "color": "darkred",
        "marginLeft": "4%",
        "marginTop": "5%",
      }
    }
  >
    Units Earned:
  </Styled(H3)>
  <Styled(Item)
    error={false}
  >
    <Styled(Input)
      keyboardType="number-pad"
      onChangeText={[Function]}
      placeholder="Enter Units"
      style={
        Object {
          "fontSize": 15,
          "marginLeft": "4%",
        }
      }
      value={4}
    />
    <Styled(Icon)
      name="checkmark-circle"
    />
  </Styled(Item)>
</Component>
`;

exports[`<PlannerForm /> renders with empty fields 1`] = `
<Component
  style={
    Object {
      "alignContent": "center",
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <Styled(H3)
    style={
      Object {
        "color": "darkred",
        "marginLeft": "4%",
        "marginTop": "5%",
      }
    }
  >
    Choose Course:
  </Styled(H3)>
  <FormPicker
    items={
      Array [
        "CSCE 2004 - Programming Foundations I",
        "CSCE 2014 - Programming Foundations II",
        "CSCE 2114 - Digital Design",
        "CSCE 2214 - Computer Organization",
        "CSCE 3193 - Programming Paradigms",
        "CSCE 3193H - Honors Programming Paradigms",
      ]
    }
    onValueChange={[Function]}
    placeholder="Select Course"
    selectedValue={[Function]}
  />
  <Styled(H3)
    style={
      Object {
        "color": "darkred",
        "marginLeft": "4%",
        "marginTop": "5%",
      }
    }
  >
    Choose Term Type:
  </Styled(H3)>
  <FormPicker
    items={
      Array [
        "Spring Intersession",
        "Spring Semester",
        "Summer Intersession",
        "Summer 10 Week Session",
        "Summer 5 Week Session 1",
        "Summer 5 Week Session 2",
        "Summer 8 Week Session",
        "Fall Intersession",
        "Fall Semester",
      ]
    }
    onValueChange={[Function]}
    placeholder="Select Term Type"
    selectedValue={[Function]}
  />
  <Styled(H3)
    style={
      Object {
        "color": "darkred",
        "marginLeft": "4%",
        "marginTop": "5%",
      }
    }
  >
    Choose Term Year:
  </Styled(H3)>
  <FormPicker
    items={
      Array [
        2000,
        2001,
        2002,
      ]
    }
    onValueChange={[Function]}
    placeholder="Select Term Year"
    selectedValue={[Function]}
  />
  <Styled(H3)
    style={
      Object {
        "color": "darkred",
        "marginLeft": "4%",
        "marginTop": "5%",
      }
    }
  >
    GPA Earned:
  </Styled(H3)>
  <Styled(Item)
    error={true}
  >
    <Styled(Input)
      keyboardType="decimal-pad"
      onChangeText={[Function]}
      placeholder="Enter Grade"
      style={
        Object {
          "fontSize": 15,
          "marginLeft": "4%",
        }
      }
      value=""
    />
    <Styled(Icon)
      name="close-circle"
    />
  </Styled(Item)>
  <Styled(H3)
    style={
      Object {
        "color": "darkred",
        "marginLeft": "4%",
        "marginTop": "5%",
      }
    }
  >
    Units Earned:
  </Styled(H3)>
  <Styled(Item)
    error={true}
  >
    <Styled(Input)
      keyboardType="number-pad"
      onChangeText={[Function]}
      placeholder="Enter Units"
      style={
        Object {
          "fontSize": 15,
          "marginLeft": "4%",
        }
      }
      value=""
    />
    <Styled(Icon)
      name="close-circle"
    />
  </Styled(Item)>
</Component>
`;
